src/stdlib_logger.f90

```diff
@@ -910,46 +949,29 @@ subroutine log_io_error( self, message, module, procedure, iostat, &
         integer :: unit
         integer :: iostat2
         character(*), parameter :: procedure_name = 'log_error'
-        character(256) :: iomsg2
+        character(256) :: iomsg2, suffix
 
-        call self % log_message( message,               &
-                                 module = module,       &
-                                 procedure = procedure, &
-                                 prefix = 'I/O ERROR' )
-
-        if ( self % units == 0 ) then
-            call write_log_io_error( output_unit )
-        else
-            do unit=1, self % units
-                call write_log_io_error( self % log_units(unit) )
-            end do
+        if ( present(iostat) ) then
+            write( suffix, '(a, i0)', err=999, iostat=iostat2, iomsg=iomsg2 ) &
+                new_line('a') // "With iostat = ", iostat
         end if
 
-    contains
-
-        subroutine write_log_io_error( unit )
-            integer, intent(in) :: unit
-
-            if ( present(iostat) ) then
-                write( unit, '("With iostat = ", i0)', err=999, &
-                    iostat=iostat2, iomsg=iomsg2 ) iostat
-            end if
-
-            if ( present(iomsg) ) then
-                if ( len_trim(iomsg) > 0 ) then
-                    call format_output_string( self,  unit,        &
-                                               'With iomsg = "' // &
-                                               trim(iomsg) // '"', &
-                                               procedure_name,     &
-                                               '    ' )
-                end if
+        if ( present(iomsg) ) then
+            if ( len_trim(iomsg) > 0 ) then
+                suffix( len_trim(suffix)+1: ) = &
+                    new_line('a') // 'With iomsg = "' // trim(iomsg) // '"'
             end if
+        end if
 
-            return
+        call self % log_message( trim(message) // trim(suffix), &
+                                 module = module,               &
+                                 procedure = procedure,         &
+                                 prefix = 'I/O ERROR' )
 
-999         call handle_write_failure( unit, procedure_name, iostat, iomsg )
+        return
 
-        end subroutine write_log_io_error
+        unit = -999
```

Comment:
Is this line valid (I mean it is just after `return` and just before `999`)?

Reply:
It should be the statement with the label 999, and the statement currently with that label should follow it.